src/components/Shop.js

```diff
@@ -1,12 +1,58 @@
 import React from 'react';
 import Product from './Product';
+import products from "./../products.json";
+import { useParams } from "react-router-dom";
 
-const Shop = (products) => {
-    return (
-        <section>
-            { products.map(p => <Product {...p} />)}
-        </section>
-    );
-}
+const Shop = (props) => {
+  const {
+    category = "",
+    sorting = "price-asc",
+    min = 0,
+    max = 100000,
+    term = "",
+  } = useParams();
+  let { main } = props;
 
-export default Shop;
\ No newline at end of file
+  let prodCat = [];
+  if (main) {
+    prodCat = products;
+  } else {
+    prodCat = products.filter((prod) => prod.category.includes(category));
+  }
+
+  prodCat = prodCat.filter(
+    (prod) =>
+      prod.price >= min && prod.price <= max && prod.description.includes(term)
+  );
```

Comment:
Trochę robi sie tu gęsto - nie do końca można zrozumieć logikę działania.
Może podziel to na funkcje, które będą bardziej sugerować co się dzieje na kolejnym etapie.